pkg/types/logs.go

```diff
@@ -95,10 +99,21 @@ func (f *AccessLogFields) KeepHeader(header string) string {
 	if f != nil && f.Headers != nil {
 		defaultValue = checkFieldHeaderValue(f.Headers.DefaultMode, defaultValue)
 
+		// Keep fast path in case of exact match.
 		if v, ok := f.Headers.Names[header]; ok {
 			return checkFieldHeaderValue(v, defaultValue)
 		}
+
+		// Check for case insensitive header name.
+		caseInsensitiveHeader := strings.ToLower(header)
+
+		for h, v := range f.Headers.Names {
```

Comment:
instead of looping over all header names, could you not do a lookup like above?

```
		if v, ok := f.Headers.Names[caseInsensitiveHeader]; ok {
			return checkFieldHeaderValue(v, defaultValue
		}
```
?

Reply:
That was my first idea, using a map for `Headers.Names` with lowercase keys, so a simple lookup in the map would be enough. It probably needs a lot more changes.

I thought that a simple solution would be enough since I was not sure that using a map would be really faster: I don't know how many headers an http request may contain (on average, or in the worst case), but I guess it's should not be too much (maybe 20 our 30 at most, so looping over 30 elements should be fast).

Would you like me to implement it using a map instead?

Reply:
That would be a good point I guess, yes :)

Reply:
@mjeanroy I totally understand that simple is better, however, this method (Keepheader()) gets called for each header in the request...So if you have 30 headers, each one is looped each time, meaning that there are 900 comparisons instead ;)

I know its a pain in the butt to think about this sort of thing, but its far easier to review and update these things early, than  trying to debug logging performance issues down the road 😄

Reply:
Hi @dtomcej, that's a very good point!
Thanks for the review, I'll do the changes asap ;)